### ISSUE TITLE
Linter does not work

### ISSUE THREAD

**Issue by mokkabonna** [AtomLinter/linter-eslint#1058]
## Issue Type
Bug

## Issue Description
Does not lint file. No errors are shown in the editor itself or on the bottom bar. However a weird behaviour is that if I run "Linter Eslint: Fix File" then it does identify and fix the fixable errors.

## Bug Checklist
- [x] Restart Atom (and reinstalled linter-eslint)
- [x] Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
- [x] Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```
Atom version: 1.22.1
linter-eslint version: 8.4.0
ESLint version: 4.12.0
Hours since last Atom restart: 0.1
Platform: win32
Using local project ESLint from: C:\Users\martinha\api_monster\project_code\monster_api\node_modules\eslint
Current file's scopes: [
  "source.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDirs": [],
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

Scopes:
```
Scopes at Cursor

source.js
```

Code:
```js
var express = require('express')
var validate = require('express-jsonschema').validate
var createProbabilitySchema = require('../schemas/generated/routes/probability/create.schema.json')
var router = express.Router()

router.post('/', validate({
  body: createProbabilitySchema
}), function(req, res) {
  res.send({})
})



module.exports = router;
```

Config:
```js
// http://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parserOptions: {
    sourceType: 'module'
  },
  env: {
    node: true,
  },
  // https://github.com/standard/standard/blob/master/docs/RULES-en.md
  extends: 'standard',
  // add your custom rules here
  'rules': {
    'space-before-function-paren': ['error', {
      'anonymous': 'ignore',
      'named': 'ignore',
      'asyncArrow': 'ignore'
    }],
    // allow paren-less arrow functions
    'arrow-parens': 0,
    // allow async-await
    'generator-star-spacing': 0,
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0
  }
}
```

CLI eslint: 
```
npm run lint
> eslint config mappings routes scripts services test utils index.js app.js

C:\Users\martinha\api_monster\project_code\monster_api\routes\probability.js
  11:1   error  More than 1 blank line not allowed  no-multiple-empty-lines
  14:24  error  Extra semicolon                     semi

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```

**Comment by mokkabonna** [comment]
Did fix it by removing a lot of packages including linter and linter-ui-default, reinstalled fixed it. Not sure exactly what action fixed it.

**Comment by skylize** [comment]
I'm guessing your Lint on Change setting for `linter` got unchecked, and when you reinstalled the setting was reset to default.

@Arcanemagus, would it be hard to pull in a few relevant settings from `linter` and `linter-ui-default` into the Debug, such as `lintOnChange`, to make it easier to debug some common problems that are technically outside the scope of `linter-eslint`?

**Comment by Arcanemagus** [comment]
> I'm guessing your Lint on Change setting for linter got unchecked, and when you reinstalled the setting was reset to default.

Uninstalling/re-installing packages has no effect on their settings as those are kept in the `config.cson`.

> @Arcanemagus, would it be hard to pull in a few relevant settings from `linter` and `linter-ui-default` into the Debug, such as `lintOnChange`, to make it easier to debug some common problems that are technically outside the scope of `linter-eslint`?

`linter` adds the `Linter: Debug` command which already shows most of the important information.
